builder/amazon/common/run_config.go

```diff
@@ -426,6 +426,18 @@ type RunConfig struct {
 	// 10m
 	WindowsPasswordTimeout time.Duration `mapstructure:"windows_password_timeout" required:"false"`
 
+	// Metadata Options
+	// See [Configure IMDS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) for details.
+	// A string to enable or disble the IMDS endpoint for an instance. Defaults to enabled.
+	// Accepts either "enabled" or "disabled"
+	HttpEndpoint string `mapstructure:"http_endpoint" required:"false"`
+	// A string to either set the use of IMDSv2 for the instance to optional or required. Defaults to "optional".
+	// Accepts either "optional" or "required"
+	HttpTokens string `mapstructure:"http_tokens" required:"false"`
+	// A numerical value to set an upper limit for the amount of hops allowed when communicating with IMDS endpoints.
+	// Defaults to 1.
+	HttpPutResponseHopLimit int64 `mapstructure:"http_put_response_hop_limit" required:"false"`
```

Comment:
I suggest exporting this into it's own struct like 

```
// Metadata Options
// See [Configure IMDS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) for details.
type MetadataOptions struct {
	// A string to enable or disble the IMDS endpoint for an instance. Defaults to enabled.
	// Accepts either "enabled" or "disabled"
	HttpEndpoint string `mapstructure:"http_endpoint" required:"false"`
	// A string to either set the use of IMDSv2 for the instance to optional or required. Defaults to "optional".
	// Accepts either "optional" or "required"
	HttpTokens string `mapstructure:"http_tokens" required:"false"`
	// A numerical value to set an upper limit for the amount of hops allowed when communicating with IMDS endpoints.
	// Defaults to 1.
	HttpPutResponseHopLimit int64 `mapstructure:"http_put_response_hop_limit" required:"false"`
}
```

Then in the RunConfig will be like 
```suggestion
Metadata MetadataOptions `mapstructure:"metadata_options" required:"false"`
```

This way you can use the `metadata_options` in the hcl2 config like: 
```
  metadata_options {
    http_tokens = "required"
    http_put_response_hop_limit = 64
  }
```

Reply:
Doing this, at the very top of the file you will have to add the MetadataOptions struct to the `//go:generate mapstructure-to-hcl2` list

Reply:
I've broken this into the struct style requested. Let me know if that looks like you expected.